### DISCUSSION TITLE
Why is the debug compilation so huge?

### DISCUSSION THREAD

**Discussion by archsolar** [bevyengine/bevy#10983] [Q&A] [answered]
Why is the debug directory 3GB when all I have is a 2d circle that moves with the mouse?
Is it supposed to be this large? I am using bevy/dynamic_linking. The initial compilation took a long time also.

![image](https://github.com/bevyengine/bevy/assets/46627284/4f612acd-1ce7-45a5-9d27-0870dadbe21b)
![image](https://github.com/bevyengine/bevy/assets/46627284/055843e2-8d97-47af-8275-8bdc4290bc50)

**Comment by MrGVSV** [comment, marked as answer]
Unfortunately that's just the [nature](https://stackoverflow.com/questions/72081919/why-is-the-target-debug-deps-directory-so-big-in-rust) of Rust programming. Rust caches the builds of all dependencies including debug symbols. Every now and then, you can try and run `cargo clean` to clear it out and build from scratch again.

**Reply by archsolar** [reply]
you're right thanks
